x-pack/plugins/security_solution/public/flyout/shared/components/flyout_navigation.tsx

```diff
@@ -37,16 +43,30 @@ export interface FlyoutNavigationProps {
    * Optional actions to be placed on the right hand side of navigation
    */
   actions?: React.ReactElement;
+  /**
+   * Boolean indicating the panel is shown in preview panel
+   */
+  isPreviewMode?: boolean;
+  /**
+   * Boolean indicating the panel is shown in rule preview
+   */
+  isPreview?: boolean;
 }
 
 /**
  * Navigation menu on the right panel only, with expand/collapse button and option to
  * pass in a list of actions to be displayed on top.
  */
 export const FlyoutNavigation: FC<FlyoutNavigationProps> = memo(
-  ({ flyoutIsExpandable = false, expandDetails, actions }) => {
+  ({ flyoutIsExpandable = false, expandDetails, actions, isPreviewMode, isPreview }) => {
     const { euiTheme } = useEuiTheme();
 
+    const history = useExpandableFlyoutHistory();
+    const isFlyoutHistoryEnabled = useIsExperimentalFeatureEnabled('flyoutHistoryEnabled');
+    const historyArray = useMemo(() => getProcessedHistory({ history, maxCount: 10 }), [history]);
```

Comment:
the `getProcessedHistory` is ran a bunch of times, sometimes when it isn't really needed. What do you think about just checking if `history.length > 1` below, passing the `history` object to the `FlyoutHistory` component and do the processing inside that `FlyoutHistory` component instead of here?

The processing only removes the last (most recent) entry, removes duplicates and revert the order so I feel like we shouldn't get into a situation where there should be a problem?
You have the value memoized so maybe I'm overthinking, but I was debugging inside the `getProcessedHistory` and it was happening a bunch of times, with a logic that stringifies then parses potential long json objects... so I was wondering if there would be a way to improve that a bit...

Reply:
Thanks! I will look into it. I was looking at another place where the history row was called more times than necessary, not sure if it is related.

Reply:
@PhilippeOberti Okay.. after taking another look, whenever we open a panel, the panel is rendered and `getProcessedHistory` is called, then the urlSync triggers the panel to load again, thus everything is run again.. 

Obviously not idea, but it appears to work according how it is built, if you see it running more times, just let me know!

```
openPanelsAction  flyout
calling getProcessedHistory
rendering flyout navigation 

urlChangedAction flyout
urlChangedAction timelineFlyout
calling getProcessedHistory
rendering flyout navigation
```